### ISSUE TITLE
JavaScript Intellisense for old style modules broken with salsa?

### ISSUE THREAD

**Issue by slowmove** [Microsoft/vscode#3813]
Since the February update where Salsa is default for JavaScript Intellisense I do not get auto completion on my old style module variables.

![skarmavbild 2016-03-08 kl 08 33 30](https://cloud.githubusercontent.com/assets/624182/13595101/0e886650-e509-11e5-9f2a-32a7b78d1fa6.png)
![skarmavbild 2016-03-08 kl 08 34 03](https://cloud.githubusercontent.com/assets/624182/13595100/0e8759a4-e509-11e5-80ae-cfad001cc447.png)

It did work before Salsa. Do I have to move to ES 6 classes and ES 6 imports?

**Comment by slowmove** [comment]
Oh I used
![skarmavbild 2016-03-08 kl 11 38 54](https://cloud.githubusercontent.com/assets/624182/13599302/72781fb6-e522-11e5-9266-2044ea04f682.png)

referencing my own files "as packages" like 
![skarmavbild 2016-03-08 kl 11 39 07](https://cloud.githubusercontent.com/assets/624182/13599316/80447cfc-e522-11e5-8642-ec6ccea9ad0f.png)

instead of have to find out the exact relative path to where I am.
This is indeed a hack and was the thing that got broken in the latest update @egamma.
Not a issue, but would be nice if it is a small fix to do =)

**Comment by jorgegonzalezperez** [comment]
Hi @slowmove 

Do you get an error if you use a function not defined in your module?

**Comment by slowmove** [comment]
@jorgegonzalezperez No I do not.

**Comment by jorgegonzalezperez** [comment]
I'm trying to configure eslint but I don't get the same behaviour

**Comment by Spown** [comment]
dunno about old/new style, TS and such. Intellisense doesn't work on very basic projects:

![](https://dl.dropboxusercontent.com/u/105338698/ShareX/2016-03/Code_2016-03-19_08-45-00.png)

explicitly calling for d.ts files by copying the declarations from automatically generated ``typings/tsd.d.ts`` shows that they are missing, while clearly present on the given location.

![](https://dl.dropboxusercontent.com/u/105338698/ShareX/2016-03/Code_2016-03-19_08-48-17.png)

adding "typings" folder path part to the inline declaration solves the problem:

![](https://dl.dropboxusercontent.com/u/105338698/ShareX/2016-03/Code_2016-03-19_09-01-28.png)

So basically the problem is VSC forgets to prefix the paths. I guess this has something to do with this new ``${workspaceRoot}`` placeholder, eh?

**Comment by egamma** [comment]
In the setup below the above scenario works, see the screenshot. So I wonder what the difference is between yours an my setup. Do the setup below work for you?

![image](https://cloud.githubusercontent.com/assets/172399/13916854/64573d28-ef5d-11e5-85a3-8cef50f89f8e.png)

The ${workspaceRoot} setting is separate from the JavaScript language service.

**Comment by egamma** [comment]
No further action planned on this issue, closing.
